src/diffTool/jsonDiff.ts

```diff
@@ -0,0 +1,383 @@
+      typedDiffs.push(typedDiff);
+    }
+  });
+  return typedDiffs;
+}
+
+/**
+ * Parse differences to the node properties
+ * @param nodeId ID of the node
+ * @param diff JSON node that has differences
+ *
+ * @returns Instance of NodeDiff for the node if there are differences
+ */
+function parseNodePropDiffs(nodeId: string, diff: object): NodeDiff {
+  let typedDiff = new NodeDiff(nodeId);
+  Object.keys(diff)
+    .filter(key => key !== JSON_LD_KEY_ID) //ignore node id
+    .forEach(key => {
+      const value = diff[key];
+
+      //check if the property of the node is an array
+      const isArray =
+        value.hasOwnProperty(JSON_DIFF_PATCH_ARRAY_KEY) &&
+        value[JSON_DIFF_PATCH_ARRAY_KEY] === JSON_DIFF_PATCH_ARRAY_VALUE;
+
+      if (isArray) {
+        //Ignore property added by jsondiffpath to indicate array property
+        Object.keys(value)
+          .filter(arrKey => arrKey !== JSON_DIFF_PATCH_ARRAY_KEY)
+          .forEach(arrKey => {
+            //Since this is flattened graph array value is either a string or a object with an id referencing other object in the graph
+            const arrValue = value[arrKey];
+            const diffType = getDiffType(arrValue);
+            ramlToolLogger.debug(
+              `Adding difference of node: ${typedDiff.id}, array property: ${key}, array key: ${arrKey}, difference type: ${DiffType[diffType]}`
+            );
+            addNodeArrayPropertyDiffs(key, arrValue, diffType, typedDiff);
+          });
+      } else {
+        const diffType = getDiffType(value);
+        ramlToolLogger.debug(
+          `Adding difference of node: ${typedDiff.id}, property: ${key}, difference type: ${DiffType[diffType]}`
+        );
+        addNodePropertyDiffs(key, value, diffType, typedDiff);
+      }
+    });
+  if (_.isEmpty(typedDiff.added) && _.isEmpty(typedDiff.removed)) {
+    //when there are just moves and no actual changes, ignore the typed diff
+    typedDiff = undefined;
+  }
+  return typedDiff;
+}
+
+/**
+ * Get the type of the difference from the difference array generated by jsondiffpatch
+ * @param diff Difference of a node or a node property
+ *
+ * @return DiffType Type of the difference
+ */
+function getDiffType(diff: Array<any>): DiffType {
+  const diffLength = diff.length;
+  if (diffLength === 1) {
+    return DiffType.ADDED;
+  }
+  if (diffLength === 2) {
+    return DiffType.MODIFIED;
+  }
+  if (diffLength === 3 && diff[2] === 0) {
+    return DiffType.REMOVED;
+  }
+  if (diffLength === 3 && diff[2] === 2) {
+    return DiffType.TEXT_DIFF;
+  }
+  if (diffLength === 3 && diff[2] === 3) {
+    return DiffType.MOVED;
+  }
+  logError(
+    `Invalid difference structure found: ${JSON.stringify(diff, null, 2)}`
+  );
+}
+
+/**
+ * Add node differences (add or remove) to NodeDiff object
+ * @param diff Node that has been added/removed
+ *
+ * @returns Instance of NodeDiff for the node
+ */
+function addNodeDiff(diff: Array<any>): NodeDiff {
```

Comment:
Is this truly `any[]` or could a more specific type be used?

Reply:
this is an object... With my recent refactor I missed to change. I will fix